### PR TITLE
Memory management!

### PR DESCRIPTION
**Pull request by olekolek1000**
malloc() and free() function

### CODE REVIEW

File: kernel/kernel.c [hunk 1]
```diff
@@ -58,10 +72,9 @@ void kmain(unsigned long magic, unsigned long multiboot_pointer) {
         }
     }
 
-    int dog = 10 / 0;
-    debug("Divide-by-zero ISR works!\n");
-    
-    //cpu_print_info();
+	mman_init();//Init memory management
```
Comment (devildefu) [inline kernel/kernel.c]: mman_init should be under init_arch

File: kernel/memory/memory.c [hunk 2]
```diff
@@ -13,8 +15,88 @@ void init_ram(void* start, void* end, uint32_t limit) {
...
+}
+
+void* malloc(size_t size){
+	size_t retry_count=0;
+	while(1){
+		char * ptr = (char*)last_address;
+		
+		mman_cell* memcell = (mman_cell*)ptr;
+		if(memcell->start == 0x00 && memcell->end == 0x00){//Memory cell is free
+			char * wanted = 0x00;
```
Comment (devildefu) [inline kernel/memory/memory.c]: char* instead of char *

File: kernel/memory/memory.c [hunk 3]
```diff
@@ -13,8 +15,88 @@ void init_ram(void* start, void* end, uint32_t limit) {
...
 }
 
-void* m_alloc(size_t size) {
-    
+//Memory manager
+const size_t mman_ptrs_start = 0x10000; //memory table starts at 64 KB
+const size_t mman_ptrs_end = mman_ptrs_start + 0x80000; //512 KB memory table
+
+size_t last_address = mman_ptrs_start;
+typedef struct{
```
Comment (devildefu) [inline kernel/memory/memory.c]: There should be a space between struct and {

Comment (olekolek1000) [reply kernel/memory/memory.c]: Done

**Pull request merged**